### PR TITLE
enable to save all workoutSetItems

### PR DESCRIPTION
**Pull request by Sho-ki**
(before)needed to push the save button every day of the week
-> (after) able to save a whole week by pushing the save button one time

### CODE REVIEW

File: src/models/workout.js [hunk 1]
```diff
@@ -125,27 +125,35 @@ const WorkoutModel = {
...
-          queryValForNew.push(val);
-        } else {
-          let { error } = await supabase
-            .from('workout_set_items')
-            .update({ set_order: item.order, reps: item.reps, sets: item.sets })
-            .match({ id: item.id });
-
-          supabaseErrorCheck(error);
-        }
+      workoutItemArray.map(async (dayItems, i) => {
```
Comment (hiroshi-tsutsui) [inline src/models/workout.js:133]: change i and j to something better.

File: src/models/workout.js [hunk 2]
```diff
@@ -125,27 +125,35 @@ const WorkoutModel = {
...
+        filteredDayItems.map(async (item, j) => {
+          if (item.id == null) {
+            let val = {
+              workout_items_id: item.workout_item_id,
+              workout_sets_id: daysIdSet[i].id,
+              set_order: j,
+              reps: item.reps,
+              sets: item.sets,
+            };
+            queryValForNew.push(val);
```
Comment (hiroshi-tsutsui) [inline src/models/workout.js:147]: `newQuery` is probably enough.

File: src/models/workout.js [hunk 3]
```diff
@@ -125,27 +125,35 @@ const WorkoutModel = {
...
-        } else {
-          let { error } = await supabase
-            .from('workout_set_items')
-            .update({ set_order: item.order, reps: item.reps, sets: item.sets })
-            .match({ id: item.id });
-
-          supabaseErrorCheck(error);
-        }
+      workoutItemArray.map(async (dayItems, i) => {
+        let filteredDayItems = dayItems.filter((setItem) => {
```
Comment (hiroshi-tsutsui) [inline src/models/workout.js:134]:
```suggestion
        let filteredDayItems = dayItems.filter((dayItem) => {
```

File: src/models/workout.js [hunk 4]
```diff
@@ -125,27 +125,35 @@ const WorkoutModel = {
...
+          return setItem.workout_item !== null;
+        });
+
+        filteredDayItems.map(async (item, j) => {
+          if (item.id == null) {
```
Comment (hiroshi-tsutsui) [inline src/models/workout.js:139]: there are too many loops here.
remove the filteredDayItems and just map through the dayItems.
```suggestion
          if (item.id === null && item.wokrout_item !== null) {
```

I think this works.

File: src/models/workout.js (hunk 1 above)
Comment (hiroshi-tsutsui) [reply src/models/workout.js:133]:
```suggestion
      workoutItemArray.map(async (workoutItem, i) => {
```

Usually when you loop through something, it becomes singular of the original data you are looping

File: src/models/workout.js [hunk 5]
```diff
@@ -125,31 +125,37 @@ const WorkoutModel = {
...
-          let { error } = await supabase
-            .from('workout_set_items')
-            .update({ set_order: item.order, reps: item.reps, sets: item.sets })
-            .match({ id: item.id });
-
-          supabaseErrorCheck(error);
-        }
+      let { data: daysIdSet } = await supabase.from('workout_sets').select().match({ users_id: userId });
+
+      let newQuery = [];
```
Comment (hiroshi-tsutsui) [inline src/models/workout.js:132]: thinking about it, can this simply just be `query`??
I don't see any `oldQuery`

File: src/models/workout.js [hunk 6]
```diff
@@ -130,32 +130,37 @@ const WorkoutModel = {
       let { data: daysIdSet } = await supabase.from('workout_sets').select().match({ users_id: userId });
 
       let query = [];
-      workoutItemSets.map(async (workoutItemSet, dayIdx) => {
-        workoutItemSet.map(async (item, orderIdx) => {
+      for (let i = 0; i < workoutItemSets.length; i++) {
+        for (let j = 0; j < workoutItemSets[i].length; j++) {
```
Comment (hiroshi-tsutsui) [inline src/models/workout.js:134]: why did you change it to for loop???

Comment (Sho-ki) [reply src/models/workout.js:134]: Because map cannot use async for the parameters. By using Promise.all(), you can use async in map, but for the readability, I thought for loop easily makes more sense.

**Review by hiroshi-tsutsui — APPROVED** [review]
lgtm
